src/Microsoft.AspNetCore.Mvc.Testing/Internal/ApplicationAssembliesStartupServicesFilter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Testing.Internal
+{
+    public class ApplicationAssembliesStartupServicesFilter : IStartupConfigureServicesFilter
+    {
+        public ApplicationAssembliesStartupServicesFilter(IList<Assembly> assemblies)
+        {
+            if (assemblies == null)
+            {
+                throw new ArgumentNullException(nameof(assemblies));
+            }
+
+            Assemblies = assemblies;
```

Comment:
`Assemblies = assemblies ?? throw new ArgumentNullException(nameof(assemblies));` for 💯 points